packages/react-core/src/components/TimePicker/TimePickerUtils.tsx

```diff
@@ -34,14 +34,13 @@ export const makeTimeOptions = (stepMinutes: number, hour12: boolean, delimiter:
   return res;
 };
 
-export const parseTime = (time: string, timeRegex: RegExp, delimiter: string, is12Hour: boolean) => {
-  time = time.trim();
+export const parseTime = (time: string | Date, timeRegex: RegExp, delimiter: string, is12Hour: boolean): string => {
   const date = new Date(time);
 
   // if default time is a ISO 8601 formatted date string, we parse it to hh:mm(am/pm) format
-  if (!isNaN(date.getDate()) && time.includes('T')) {
+  if (!isNaN(date.getDate()) && (time instanceof Date || time.includes('T'))) {
```

Comment:
Why the `time.includes('T')` check?

Reply:
You may have a better solution, but it's because ISO 8601 strings which include a time always have a T in them to delineate the time. 

if you pass any string of numbers to a new Date(<number>) it will return a valid date that is not reflective of the intended time. It basically is checking that the passed in time is either a valid Date object, or it is a string of numbers that is written in ISO 8601 format. In either of those cases, it's parsed using javascript date functions. Otherwise, it's parsed as a string.

Reply:
Ah I didn't notice you changed the function signature to `time: string | Date`.